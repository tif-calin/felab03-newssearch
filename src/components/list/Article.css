@import url('https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,200;0,300;0,400;0,600;0,700;0,900;1,200;1,300;1,400;1,600;1,700;1,900&display=swap');

.Article {
  font-family: 'Source Sans Pro', sans-serif;
}

.Article > span:first-child {
  font-weight: 500;
  font-size: 1.1rem;
}

.Article > span:nth-child(2) {
  font-weight: 200;
}

.Article > p {
  margin: 1rem;
  margin-top: 0.25rem;
  font-size: 0.9rem;
  font-weight: 300;
}
