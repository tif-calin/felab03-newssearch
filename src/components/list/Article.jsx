import React from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

const Article = props => {
  
  return <li 
    className={styles.Article}
    data-testid="article"
  >
    <span>{props.article.title}</span> <span>|| {props.article.author}</span>
    <p>{props.article.description}</p>
  </li>;
};

Article.propTypes = {
  article: PropTypes.object.isRequired
};

export default Article;
