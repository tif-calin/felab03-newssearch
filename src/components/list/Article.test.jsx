import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());

  it('renders Article', () => {
    const { asFragment } = render(<Article article={{ 
      title: 'Florida man is just misunderstood', 
      description: 'He\'s actually not a bad guy at all.',
      author: 'Florida Man Blog'
    }}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
