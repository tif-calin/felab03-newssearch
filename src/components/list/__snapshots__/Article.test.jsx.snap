// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <li
    class="Article"
    data-testid="article"
  >
    <span>
      Florida man is just misunderstood
    </span>
     
    <span>
      || Florida Man Blog
    </span>
    <p>
      He's actually not a bad guy at all.
    </p>
  </li>
</DocumentFragment>
`;
